crates/ruff_linter/src/linter.rs

```diff
@@ -40,11 +40,48 @@ use crate::{directives, fs, Locator};
 pub struct LinterResult {
     /// A collection of diagnostic messages generated by the linter.
     pub messages: Vec<Message>,
-    /// A flag indicating the presence of syntax errors in the source file.
-    /// If `true`, at least one syntax error was detected in the source file.
+    /// Flag indicating that the parsed source code does not contain any
+    /// [`ParseError`]s
+    has_valid_syntax: bool,
+    /// Flag indicating that the parsed source code does not contain any
+    /// [`UnsupportedSyntaxError`]s
+    has_no_unsupported_syntax_errors: bool,
+}
+
+impl LinterResult {
```

Comment:
It's a bit unfortunate that these methods need to be repeated on both `Parsed` and `LinterResult`. We could explore another approach to avoid duplication, not necessarily in this PR, that utilizes `bitflags` for these two booleans like `SyntaxFlags` that can then be queried to ask whether there are parse errors or unsupported syntax errors.